io/disk/arcconf/arcconf_migration.py

```diff
@@ -0,0 +1,250 @@
+        disk_val2 = ""
+        # create the inital disk required for raid creation
+        loop_count = 0
+        for disk in self.disk_no:
+            loop_count += 1
+            disk_val1 += " %s %s" % (self.channel_no, disk.strip())
+            if loop_count == 2 and self.initial_raid in ('0', '1'):
+                break
+            if loop_count == 3 and self.initial_raid == '5':
+                break
+            if loop_count == 4 and self.initial_raid in ('6', '10'):
+                break
+        # create the disk required for migration
+        loop_count = 0
+        for disk in self.disk_no:
+            loop_count += 1
+            disk_val2 += " %s %s" % (self.channel_no, disk.strip())
+            if loop_count == int(self.disk_migrated):
+                break
+        # Start migration
+        for migrate in self.migrate_raid:
+            # create logical drive
+            self.log.info("Creating RAID %s" % self.initial_raid)
+            cmd = "echo y | arcconf create %s logicaldrive %s %s %s"\
+                  % (self.crtl_no, self.disk_size,
+                     self.initial_raid, disk_val1)
+            self.check_pass(cmd, "Failed to run %s" % cmd)
+
+            time.sleep(int(self.migration_sleep))
+
+            # rebuild tests
+            if self.spare_drive != "" and self.initial_raid == '1':
+                cmd = "echo y | arcconf setstate %s device %s %s HSP" % \
+                      (self.crtl_no, self.channel_no, self.spare_drive)
+                self.check_pass(cmd, "Failed to run %s" % cmd)
+
+                time.sleep(int(self.migration_sleep))
+
+                cmd = "echo y | arcconf setstate %s device %s %s DDD" % \
+                      (self.crtl_no, self.channel_no, self.disk_no[1])
+                self.check_pass(cmd, "Failed to run %s" % cmd)
+
+                cmd = "arcconf getconfig %s LD" % (self.crtl_no)
+                self.check_pass(cmd, "Logical drive getconfig failed")
+
+                time.sleep(int(self.migration_sleep))
+
+                cmd = "echo y | arcconf setstate %s device %s %s RDY " % \
+                      (self.crtl_no, self.channel_no, self.disk_no[1])
+                self.check_pass(cmd, "Failed to run %s" % cmd)
+
+            # Migration tests
+            else:
+                self.log.info("Migrating from RAID %s to %s" %
+                              (self.initial_raid, migrate))
+                cmd = "echo y | arcconf MODIFY %s FROM %s TO %s %s %s" % \
+                      (self.crtl_no, self.logicaldrive,
+                       self.disk_size, migrate, disk_val2)
+                self.check_pass(cmd, "Failed to run %s" % cmd)
+
+            time.sleep(int(self.migration_sleep))
+
+            cmd = "arcconf getconfig %s LD" % (self.crtl_no)
+            self.check_pass(cmd, "Logical drive getconfig failed")
+
+            cmd = "lsscsi | grep LogicalDrv | awk \'{print $7}\'"
+            drive = self.cmdop_list(cmd)
```

Comment:
hmm okay, Done. made modification to exsisting code itself as below.


    def cmdop_list(self, cmd):
        """
        Function returns the output of a command
        """
        if cmd == 'OS':
            cmd = "lsscsi  | grep LogicalDrv | awk \'{print $7}\'"
        val = process.run(cmd, shell=True, ignore_status=True,
                          allow_output_check='stdout')
        if val.exit_status:
            self.fail("cmd %s Failed" % (cmd))
        return val.stdout.rstrip()


and will be calling as self.cmdop_list("OS")